ideasbox/settings.py

```diff
@@ -1,161 +1,25 @@
-
-TIME_ZONE = os.environ.get('TIME_ZONE', 'UTC')
-
-USE_I18N = True
-
-USE_L10N = True
-
-USE_TZ = True
-
-AVAILABLE_LANGUAGES = (
-    ('en', _('English')),
-    ('fr', _('French')),
-    ('ar', _('Arabic')),
-)
-
-SUPPORTED_LANGUAGES = os.environ.get('SUPPORTED_LANGUAGES', 'fr en ar').split()
-LANGUAGES = []
-for code, label in AVAILABLE_LANGUAGES:
-    if code in SUPPORTED_LANGUAGES:
-        LANGUAGES.append((code, label))
-
-LOCALE_PATHS = (
-    os.path.join(BASE_DIR, 'locale'),
-)
-
-# Static files (CSS, JavaScript, Images)
-# https://docs.djangoproject.com/en/1.7/howto/static-files/
-
-STATIC_URL = '/static/'
-MEDIA_URL = '/media/'
-
-
-# Ideas Box specifics
+# Every box will have some edge specific needs, such as a specific user model,
+# we manage this with per box settings, but we want those specific settings
+# to be versionned, for two reasons: easier to debug when there is no hidden
+# local config, and easier to manage code upgrade.
 try:
-    IDEASBOX_ID = os.environ['IDEASBOX_ID']
-except KeyError:
-    if DEBUG:
-        IDEASBOX_ID = 'debugbox'
-    else:
-        raise
-try:
-    STORAGE_ROOT = os.environ['DATASTORAGE']
-except KeyError:
-    if DEBUG:
-        try:
-            os.makedirs('storage/main')
-        except OSError:
-            pass
-        STORAGE_ROOT = os.path.join(BASE_DIR, 'storage')
-    else:
-        raise
-MEDIA_ROOT = os.path.join(STORAGE_ROOT, 'main')
-STATIC_ROOT = os.path.join(STORAGE_ROOT, 'static')  # TODO move out of backuped
-                                                    # storage
-AUTH_USER_MODEL = os.environ.get('AUTH_USER_MODEL', 'ideasbox.DefaultUser')
-IDEASBOX_NAME = os.environ.get('IDEASBOX_NAME', IDEASBOX_ID)
-
-# Database
-# https://docs.djangoproject.com/en/1.7/ref/settings/#databases
-
-DATABASES = {
-    'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(MEDIA_ROOT, 'default.sqlite'),
-    }
-}
-
-SERVICES = [
-    {'name': 'apache2', 'description': _('Daemon which provide web content')},
-    {'name': 'bind9', 'description': _('Daemon which provide local DNS')},
-    {'name': 'kalite',
-        'description': _('Daemon which provide KhanAcademy on lan')},
-    {'name': 'kiwix',
-        'description': _('Daemon which provide Wikipedia on lan')},
-    {'name': 'ntp', 'description': _('Net time protocol')},
-    {'name': 'ssh',
-        'description': _('Daemon used for distant connexion to server')},
-]
+    sub = importlib.import_module(".conf." + IDEASBOX_ID, package="ideasbox")
+except ImportError:
+    from .conf import base as sub
+finally:
+    SETTINGS_MODULE = sub.__name__  # Make if available as a settings, to
+                                    # be able to display it in the admin
+    sys.stdout.write('Importing settings from %s\n' % SETTINGS_MODULE)
+    ldict = locals()
+    for k in sub.__dict__:
+        if k.isupper() and not k.startswith('__') or not k.endswith('__'):
+            ldict[k] = sub.__dict__[k]
```

Comment:
I understand your point, and I may be thick, but really I'm convinced this isn't the right place to do it (now that you kind of convinced me you need to do it).

Settings are a place to set settings, not dynamically do anything. I'd recommend doing this hackery (which will thus be much more limited and much simpler) in the `manage.py`, using the hostname to choose what to store in the `DJANGO_SETTINGS_MODULE` (this way it'll simply be a string interpolation, instead of this module import fiddling).

What do you think?